src/containers/LoginForm.test.js

```diff
@@ -1,13 +1,246 @@
+
+    it('should invoke handleInputChange on change within email input field',
+      () => {
+        const mockEvent = { target: { name: 'testName', value: 'testValue' } }
+        wrapper.instance().handleInputChange = jest.fn();
+        wrapper.find('#email-input').simulate('change', mockEvent);
+        expect(wrapper.instance().handleInputChange)
+          .toHaveBeenCalledWith(mockEvent);
+
+        wrapper.find('#password-input').simulate('change', mockEvent);
+        expect(wrapper.instance().handleInputChange)
+          .toHaveBeenCalledWith(mockEvent);
+    });
+
+    it('should invoke validateInputs when handleSubmit is called', () => {
+      const mockEvent = { preventDefault: jest.fn() };
+      wrapper.instance().validateInputs = jest.fn();
+      wrapper.instance().handleSubmit(mockEvent);
+      expect(wrapper.instance().validateInputs).toHaveBeenCalled();
+    });
+
+    it('should setState when validateInputs is called with an input empty',
+      () => {
+        const mockStartingState = {email: 'fakeemail@email.com',
+          password: '',
+          error: ''};
+        const mockEndingState = {email: 'fakeemail@email.com',
+          password: '',
+          error: 'All input fields required'};
+        wrapper.setState(mockStartingState);
+        expect(wrapper.state()).toEqual(mockStartingState);
+
+        wrapper.instance().validateInputs();
+        expect(wrapper.state()).toEqual(mockEndingState);
+    });
+
+    it('should setState when handleInputChange is called', () => {
+      const mockEvent = { target: { name: 'email', value: 'testValue' } };
+      const mockStartingState = {
+        email: 'fakeemail@email.com',
+        password: 'fakepassword',
+        error: 'fakeerror' };
+      const mockEndingState = {
+        email: 'testValue',
+        password: 'fakepassword',
+        error: '' };
+      wrapper.setState(mockStartingState);
+      wrapper.instance().handleInputChange(mockEvent);
+      expect(wrapper.state()).toEqual(mockEndingState);
+    });
+
+    it('should invoke handleError prop method when handleInputChange is called',
+      () => {
+        const mockEvent = { target: { name: 'email', value: 'testValue' } };
+        wrapper.instance().handleInputChange(mockEvent);
+        expect(mockHandleError).toHaveBeenCalled();
+    });
+
+    it('should invoke fetchUserLogin when handleSubmit is called', () => {
+      const mockStartingState = {
+        email: 'Marge@turing.io',
+        password: 'fakepassword',
+        error: '' };
+      wrapper.setState(mockStartingState);
+
+      const mockEvent = { preventDefault: jest.fn() };
+      wrapper.instance().handleSubmit(mockEvent);
+      expect(fetchUserLogin)
+        .toHaveBeenCalledWith('Marge@turing.io', 'fakepassword');
+    });
+
+    it('should invoke fetchRatings when fetchUserLogin resolves', () => {
+      fetchUserLogin('Marge@turing.io', 'fakepassword');
+      expect(fetchRatings).toHaveBeenCalledWith(9);
+    });
```

Comment:
Requesting your review on this test.  I ran it with `8` instead of `9` on line 156 to make sure if failed and it showed that it was being called 2 times.  I'm not sure if one is from the test above it?  This is what the failure looked like:  (Again, its ok that it failed, thats what I was checking for - but why is it showing `Number of Calls 2`?
![Screen Shot 2019-12-31 at 9 13 49 PM](https://user-images.githubusercontent.com/48163945/71638080-431a0180-2c13-11ea-85d7-a7a3911e7ad6.jpg)